app/views/posts/specific_indexing.html.slim

```diff
@@ -29,3 +37,16 @@ h1.display-4.text-center.font-weight-bold.mt-5 All posts
 
       = link_to 'Delete', post_path(post), method: :delete, data: { confirm: "Are you sure?" }, class: 'btn btn-sm btn-danger float-right'
       = link_to 'Show', post_path(post), class: 'btn btn-sm btn-info float-right mr-3'
+
+      - if post.donator == nil && current_user.role != 'admin'
```

Comment:
I can see such condition on many places, Please move these to helpers, and use helper functions.

Reply:
Noted